### DISCUSSION TITLE
LLM + optimization engines

### DISCUSSION THREAD

**Discussion by Jacck** [modelcontextprotocol/modelcontextprotocol#1186] [General]
### Pre-submission Checklist

- [X] I have verified this would not be more appropriate as a feature request in a specific repository
- [X] I have searched existing discussions to avoid duplicates

### Your Idea

Recently I developed mcp-ortools server, integrating Google OR-Tools with MCP. Here is: github: https://github.com/Jacck/mcp-ortools 

So far I have seen only one talk about combining LLMs and optimization engines: https://www.youtube.com/live/wm9-7VBpdEo?si=suSMCQF9Ytrssgvf

I'm curious whats your view on the perspective of formulating problem in natural language using LLMs, sending model to opt engine for solution, and interpreting results using LLMs. I believe that there are some use cases worth to develop.

### Scope

- [ ] Protocol Specification
- [ ] SDK Features
- [ ] Documentation
- [ ] Developer Experience
- [X] Other

**Comment by vincentkoc** [comment]
@Jacck I'm a maintainer for Opik (https://github.com/comet-ml/opik) and recently been working on Opik Optimizer SDK which is an open source SDK for LLM optimizers. I have recently introduced the ability to [optimize MCP tool signatures](https://www.comet.com/docs/opik/agent_optimization/algorithms/tool_optimization#2-mcp-model-context-protocol-tools---beta) as well

**Comment by localden** [comment]
We've limited Discussions in this repository to meeting notes and are closing this thread. To continue the conversation, pick a channel from [Contributor Communication](https://modelcontextprotocol.io/community/communication). Thanks for posting.
